FF/Pressure.C

```diff
@@ -18,11 +18,27 @@ void preciceAdapter::FF::Pressure::write(double* buffer, bool meshConnectivity,
 
     if (this->locationType_ == LocationType::volume)
     {
-        // set in Interface::addCouplingDataR/W
-        forAll(p_->ref(), i)
+        if(cellSetNames_.empty())
         {
-            buffer[bufferIndex++] = p_->ref()[i];
+            forAll(p_->ref(), i)
+            {
+                buffer[bufferIndex++] = p_->ref()[i];
+            }
         }
+        else
+        {
+            for (auto cellSetName : cellSetNames_) {
```

Comment:
Here, and in similar places where you iterate over a container to only read values, you could default to `const auto &`, for safety and performance reasons (at least in the parts of the code called in every coupling time window).

```suggestion
            for (const auto & cellSetName : cellSetNames_) {
```